apps/olat-api/src/index.ts

```diff
@@ -0,0 +1,406 @@
+  )
+}
+
+app.get('/api/configuration/activityTypes', (req: Request, res: Response) => {
+  const headerStatusCode = checkHeader(req)
+  if (headerStatusCode !== StatusCode.SUCCESS) {
+    return res
+      .status(headerStatusCode)
+      .json({ error: 'Invalid request headers' })
+  }
+
+  getActivityTypes().then((activityTypes) => {
+    if (activityTypes === null) {
+      return res
+        .status(StatusCode.NOT_FOUND)
+        .json({ error: 'Course not found' })
+    }
+
+    res.set('Content-Type', 'application/json')
+    res.status(StatusCode.SUCCESS).json({
+      activityTypes: activityTypes,
+      timestamp: new Date().toISOString(),
+      api: API_NAME,
+    })
+  })
+})
+
+async function getCourseActivityTypes(courseId: string): Promise<any[] | null> {
+  const course = await prisma.course.findUnique({
+    where: { id: courseId },
+    include: {
+      // NOTE: modify if required
+      liveQuizzes: true,
+      practiceQuizzes: true,
+      microLearnings: true,
+    },
+  })
+  if (!course) return null
+
+  const isGamificationEnabled = course.isGamificationEnabled
+  const liveQuizzes = course.liveQuizzes ?? []
+  const practiceQuizzes = course.liveQuizzes ?? []
+  const microLearnings = course.liveQuizzes ?? []
+
+  const mapOverview: Record<string, any[]> = {
+    // NOTE: modify if required
+    'live-quizzes': liveQuizzes,
+    'practice-quizzes': practiceQuizzes,
+    'micro-learnings': microLearnings,
+  }
+
+  const mapSubselection: Record<string, any[]> = {
+    // NOTE: modify if required
+    'live-quiz': liveQuizzes,
+    'practice-quiz': practiceQuizzes,
+    'micro-learning': microLearnings,
+  }
+  const activityKeysGamification = ['leaderboard'] // NOTE: modify if required
+
+  const activityTypes = activityTypesAvailable
+    .map(
+      ({
+        id,
+        title,
+        path,
+        olatConfigurationKey,
+        isSubselectionRequired,
+        isEmailTransferRequired,
+      }) => {
+        if (olatConfigurationKey in mapOverview) {
+          const count = mapOverview[olatConfigurationKey]?.length || 0
+          const newTitle = title + ` (${count})`
+          return { id, newTitle, olatConfigurationKey, isSubselectionRequired }
+        } else if (olatConfigurationKey in mapSubselection) {
+          const count = mapSubselection[olatConfigurationKey]?.length || 0
+          return count > 0
+            ? { id, title, olatConfigurationKey, isSubselectionRequired }
+            : undefined
+        } else if (activityKeysGamification.includes(olatConfigurationKey)) {
+          return isGamificationEnabled
+            ? { id, title, path, olatConfigurationKey, isSubselectionRequired }
+            : undefined
+        } else {
+          return {
+            id,
+            title,
+            path,
+            olatConfigurationKey,
+            isSubselectionRequired,
+          }
+        }
+      }
+    )
+    .filter((activityType) => activityType !== undefined)
```

Comment:
idea: we could improve the efficiency of this by using reduce to avoid looping twice